jme3-effects/src/main/java/com/jme3/post/filters/SsrFilter.java

```diff
@@ -0,0 +1,305 @@
+    private Pass ssrPass;
+    private boolean approximateNormals = false;
+    private boolean approximateGlossiness = true;
+
+    private float downSampleFactor = 1f;
+
+    private Material ssrMaterial;
+    private boolean fastBlur = false;
+    private int raySteps = 16;
+    private boolean sampleNearby = true;
+    private float stepLength = 1.0f;
+    private float blurScale = 1f;
+    private float sigma = 5f;
+    private float reflectionFactor = 1f;
+    private Vector2f nearFade = new Vector2f(0.01f, 1.0f);
+    private Vector2f farFade = new Vector2f(200f, 300f);
+
+
+    public SsrFilter(){
+        super("SSR Filter");
+    }
+    
+    @Override
+    protected boolean isRequiresDepthTexture() {
+        return true;
+    }
+
+    @Override
+    protected void initFilter(AssetManager manager, RenderManager renderManager, ViewPort vp, int w, int h) {
+        
+        this.renderManager = renderManager;
+        this.viewPort = vp;
+        int screenWidth = w;
+        int screenHeight = h;
+                postRenderPasses = new ArrayList<>();
+
+
+        frustumNearFar = new Vector2f();
+
+        float farY = (vp.getCamera().getFrustumTop() / vp.getCamera().getFrustumNear()) * vp.getCamera().getFrustumFar();
+        float farX = farY * ((float) screenWidth / (float) screenHeight);
+        frustumNearFar.x = vp.getCamera().getFrustumNear();
+        frustumNearFar.y = vp.getCamera().getFrustumFar();
+        
+        
+//        if(!approximateNormals){
+            normalPass = new Pass();
+            normalPass.init(renderManager.getRenderer(), (int) (screenWidth / downSampleFactor), (int) (screenHeight / downSampleFactor), Image.Format.RGBA8, Image.Format.Depth);
+//        }
+
+        ssrMaterial = new Material(manager, "Common/MatDefs/SSR/ssr.j3md"); 
+//        if(!approximateNormals){
+            ssrMaterial.setTexture("Normals", normalPass.getRenderedTexture());
+//        }
+        ssrMaterial.setInt("RaySamples", raySteps);
+        ssrMaterial.setInt("NearbySamples", sampleNearby ? 4 : 0);
+        ssrMaterial.setFloat("StepLength", stepLength);
+        ssrMaterial.setFloat("ReflectionFactor", reflectionFactor);
+        ssrMaterial.setBoolean("ApproximateNormals", approximateNormals);
+        ssrMaterial.setVector2("NearReflectionsFade", nearFade);
+        ssrMaterial.setVector2("FarReflectionsFade", farFade);
+        
+        ssrPass = new Pass("SSR pass") {
+
+            @Override
+            public boolean requiresDepthAsTexture() {
+                return true;
+            }
+
+            @Override
+            public boolean requiresSceneAsTexture() {
+                return true; //To change body of generated methods, choose Tools | Templates.
+            }
+            
+            
+        };
+
+        ssrPass.init(renderManager.getRenderer(), (int) (screenWidth / downSampleFactor), (int) (screenHeight / downSampleFactor), Image.Format.RGBA8, Image.Format.Depth, 1, ssrMaterial);
```

Comment:
The format shouldn't be forced to RGBA8.

Reply:
How do you propose doing it? A setter for the Image.Format? Or is it just that RGBA8 isn't the supposed format?

Reply:
Mhh.. yes a setter and maybe use RGBA16F as default, since the default format for filters is RGB111110F .